frontend/src/components/dashboard/create/Create.tsx

```diff
@@ -16,17 +22,17 @@ const Create = () => {
   const [campaign, setCampaign] = useState(new Campaign({}))
   const [isLoading, setLoading] = useState(true)
 
-  async function loadProject() {
+  const loadProject = useCallback(async () => {
     if (id) {
       const campaign = await getCampaignDetails(+id)
       setCampaign(campaign)
       setLoading(false)
     }
-  }
+  }, [id])
 
   useEffect(() => {
     loadProject()
-  }, [id])
+  }, [loadProject])
```

Comment:
can we just use useEffect with id as dependency, and loadProject to take in an id as arg 
useCallback is more appropriate when function is passed to a child component as a prop and you dont want it to rerender each time cos of referential inequality.

Reply:
same goes for the other useCallback changes